lib/cartopy/io/img_tiles.py

```diff
@@ -320,6 +320,57 @@ def _image_url(self, tile):
         return f'https://a.tile.openstreetmap.org/{z}/{x}/{y}.png'
 
 
+class StadiaMapsTiles(GoogleWTS):
+    """
+    Retrieves tiles from stadiamaps.com.
+
+    For a full reference on the styles available please see
+    https://docs.stadiamaps.com/themes/. A few of the specific styles
+    that are made available are ``alidade_smooth``, ``stamen_terrain`` and
+    ``osm_bright``.
+
+    Using the Stadia Maps API requires including an attribution. Please see
+    https://docs.stadiamaps.com/attribution/ for details.
+
+    For most styles that means including the following attribution:
+
+    `© Stadia Maps <https://www.stadiamaps.com/>`_
+    `© OpenMapTiles <https://openmaptiles.org/>`_
+    `© OpenStreetMap contributors <https://www.openstreetmap.org/about/>`_
+
+    with Stamen styles *additionally* requiring the following attribution:
+
+    `© Stamen Design <https://stamen.com/>`_
+
+    Parameters
+    ----------
+    apikey : str, required
+        The authentication key provided by Stadia Maps to query their APIs
+    style : str, optional
+        Name of the desired style. Defaults to ``alidade_smooth``.
+        See https://docs.stadiamaps.com/themes/ for a full list of styles.
+    cache : bool or str, optional
+        If True, the default cache directory is used. If False, no cache is
+        used. If a string, the string is used as the path to the cache.
+    """
+
+    def __init__(self, apikey, style='alidade_smooth', cache=False):
+        super().__init__(cache=cache)
+        self.apikey = apikey
+        if style in ("alidade_satellite", "stamen_watercolor"):
```

Comment:
Some notes on `alidade_satellite` that'd be really easy to miss in our docs...

1. We don't actually support this as a full rendered style at the moment (this style ID won't actually work).
2. The JPGs that do exist under a slightly different URL are for the raw imagery imagery only and are not a standard 256x256 tile but rather @2x 512 tiles, which are intended to be used in conjuction with a vector style.

The terms on this raw imagery are also a little different per our upstream provider (some details here: https://docs.stadiamaps.com/map-styles/alidade-satellite/), so I TL;DR I would suggest _not_ adding this condition for the satellite style or mentioning it. We do intend to add this as a "normal" raster style in the future with the other layers on top, but it's probably best to just remove this for now.

Reply:
Thanks for the info, I removed the satellite option for now.